### PR TITLE
test: enables service tests

### PR DESCRIPTION
**Pull request by epau**
This PR enables service tests. Previously, they were disabled due to an issue with CRT. 
CRT would hang on a call to cleanup, because CRT objects were still being held in memory by the SDKDefaultIO singleton. 

Additionally, this PR removes the following service tests.
- S3 **HeadObjectErrorTest** and **HeadBucketErrorTests**. These tests were not accurate. An issue has been created to create new tests that should take their place https://github.com/awslabs/aws-sdk-swift/issues/886
- S3 **PutObjectTests**. These tests are not actually testing any business logic. They are too coupled to the test configuration and seem to just be testing the details in the test. 
- Route53 tests. Similar to the S3 tests, these aren't actually testing any business logic and appear to just be testing the operation stack configuration in the test itself. 

We should prioritize implementing interceptors as interceptors should be the mechanism used by service tests to verify behavior. For example, to test that a certain operation decodes a specific response successfully, we should add an interceptor after the deserialization step that verifies the deserialized output is as we expect. Currently, we duplicate the entire operation stack in the test which is pointless as it doesn't actually test the specific operation.

## Issue \#
#814 

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ganeshnj** [comment]
per https://www.conventionalcommits.org/en/v1.0.0/#specification this seems to fit more into `test` category

File: Tests/Services/AWSRoute53Tests/ListResourceRecordSetsRequestTest.swift [hunk 1]
```diff
@@ -1,65 +0,0 @@
-// Code generated by smithy-swift-codegen. DO NOT EDIT!
-
-@testable import AWSRoute53
```
Comment (ganeshnj) [inline Tests/Services/AWSRoute53Tests/ListResourceRecordSetsRequestTest.swift:3]: do we have a backlog to add it back?

File: codegen/sdk-codegen/aws-models-test/s3/s3-tests.smithy [hunk 2]
```diff
@@ -1,83 +0,0 @@
-$version: "1.0"
```
Comment (ganeshnj) [inline codegen/sdk-codegen/aws-models-test/s3/s3-tests.smithy:1]: we must add a backlog, ideally a TODO in kt codebase.

**Review by ganeshnj — APPROVED** [review]

File: Tests/Services/AWSRoute53Tests/ListResourceRecordSetsRequestTest.swift (hunk 1 above)
Comment (epau) [reply Tests/Services/AWSRoute53Tests/ListResourceRecordSetsRequestTest.swift:3]: Added one here https://github.com/awslabs/aws-sdk-swift/issues/888

**Pull request merged**